### PR TITLE
141 build workflow fails due to missing datatbases

### LINKED ISSUE #141: `build` workflow fails due to missing datatbases
As verbose logs indicate, the current reason for the failure of the build command ([log](https://github.com/BIMSBbioinfo/pigx_sars-cov-2/runs/8122898107)) is due to databases not being found within the runner. 

## Proposed solution

Adding snakemake rules for dastabase downloads. These should be optional and be controlled via the settings file, but be active by default.

### PR DESCRIPTION
**Pull request by jonasfreimuth**
Closes #141.
(This is just the branch get-build-to-work)

### CODE REVIEW

**Comment by jonasfreimuth** [comment]
Can be merged once the right database archives are on bimsbstatic.

File: tests/settings.yaml [hunk 1]
```diff
@@ -5,9 +5,15 @@ locations:
...
-  kraken-db-dir: /local/Projects/AAkalin_pathogenomics/pigx_sars-cov-2_project/databases/kraken_db
-  krona-db-dir: /local/Projects/AAkalin_pathogenomics/pigx_sars-cov-2_project/databases/krona_db
-  vep-db-dir: /local/Projects/AAkalin_pathogenomics/pigx_sars-cov-2_project/databases/vep_db
+
+databases:
+  kraken2:
+    archive-url: bimsbstatic.mdc-berlin.de/akalin/AAkalin_pathogenomics/databases_small/kraken_db.tar.gz
+  krona:
+    use-prebuilt: true
+    archive-url: bimsbstatic.mdc-berlin.de/akalin/AAkalin_pathogenomics/databases_small/krona_db.tar.gz
```
Comment (rekado) [inline tests/settings.yaml:14]: Shouldn't these URLs have a protocol prefix (such as https://)?

File: snakefile.py [hunk 2]
```diff
@@ -522,6 +549,38 @@ def render_qc_report_params(wildcards, input, output = None, threads = None, res
             for name in generated:
                 logger.info("  - {}".format(name))
 
+
+# ANNOT: If the any of the databases is not present, download it to the location
+# specified in the settings file.
+rule download_kraken_db:
+    output: directory(KRAKEN_DB)
```
Comment (rekado) [inline snakefile.py:556]: Should these be marked as `ancient`?  Either here or wherever they are used?

**Review by rekado — APPROVED** [review]

File: tests/settings.yaml (hunk 1 above)
Comment (jonasfreimuth) [reply tests/settings.yaml:14]: With the latest commits, urls without protocol will be assumed to be http.

File: snakefile.py (hunk 2 above)
Comment (jonasfreimuth) [reply snakefile.py:556]: Not sure about that... Ancient is only used to denote files that should always be assumed to be older than outputs and will therefore not trigger a rerun if they are updated. But if the databases are already there, they are older, if they are newly downloaded they will also already be older. However if the databases are updated, we do want them to trigger a rerun, so I don't see why we would need an ancient tag there. This is the first time I am thinking about the concept of the ancient tag though...

**Pull request merged**